internal/langserver/handlers/hooks_module.go

```diff
@@ -44,6 +44,25 @@ func moduleTelemetryData(mod *state.Module, ch state.ModuleChanges, store *state
 	if len(mod.Meta.CoreRequirements) > 0 {
 		properties["tfRequirements"] = mod.Meta.CoreRequirements.String()
 	}
+	if mod.Meta.Cloud != nil {
+		properties["cloud"] = true
+
+		hostname := mod.Meta.Cloud.Hostname
+
+		// https://developer.hashic orp.com/terraform/language/settings/terraform-cloud#usage-example
+		// Required for Terraform Enterprise;
+		// Defaults to app.terraform.io for Terraform Cloud
+		if hostname == "" {
+			hostname = "app.terraform.io"
+		}
+
+		// anonymize any non-default hostnames
+		if hostname != "app.terraform.io" {
+			hostname = "custom-hostname"
+		}
```

Comment:
This looks slightly inconsistent with how we check & report hostnames below for backends.

I don't have too strong opinion on whether we should be reporting the difference between default and empty (implied default) hostname, but I do think we should approach both the same way.

Reply:
According to https://developer.hashicorp.com/terraform/cli/cloud/settings#hostname, hostname is optional, and is an empty string if not specified. Using the same logic as backend would result is us sending an empty string in telemetry, which defeats the purpose in doing this.

Reply:
I would assume that telemetry metrics with empty string still get sent through? As in - we'd still be able to see any `cloud` blocks parsed, the only difficulty would be treating empty strings as `app.terraform.io` when processing the data.

To put it differently - I'm happy with your implementation - which implies that we do no further interpretation of empty strings in AppInsights, but in that case I think we should do the same for the remote `backend` below, otherwise we end up comparing apples to pears.

Reply:
Yes, sending empty data is allowed.

If the default is known, I don't see the need for sending empty strings and then having to do conversion when reporting.

I'll open a ticket for backend reporting.